llm/nemorl/nemorl.sky.yaml

```diff
@@ -48,17 +48,25 @@ run: |
   cd nemo-rl
   head_ip=`echo "$SKYPILOT_NODE_IPS" | head -n1`
   num_nodes=`echo "$SKYPILOT_NODE_IPS" | wc -l`
+  # Isolate inner Ray from outer Ray env
+  unset RAY_RAYLET_PID
+  export RAY_ADDRESS=$head_ip:6379
   if [ "$SKYPILOT_NODE_RANK" == "0" ]; then
-    ps aux | grep ray | grep 6379 &> /dev/null || uv run ray start --head  --disable-usage-stats --port 6379
-    sleep 5
+    ps aux | grep ray | grep 6379 &> /dev/null || uv run ray start --head --disable-usage-stats --port 6379
+    # Wait for all nodes to join
+    timeout=180
+    until uv run python -c 'import os,ray,sys; ray.init(address=os.environ["RAY_ADDRESS"]); sys.exit(0 if sum(1 for n in ray.nodes() if n["Alive"])==int(os.environ["SKYPILOT_NUM_NODES"]) else 1)'; do
+      sleep 3; timeout=$((timeout-3)); if [ $timeout -le 0 ]; then echo "Ray nodes not ready"; exit 1; fi
+    done
```

Comment:
We should add a similar snippet to the ray example: https://github.com/skypilot-org/skypilot/blob/master/examples/distributed_ray_train/ray_train.yaml

Reply:
Removed this snippet to keep things minimal.